common/src/main/java/io/netty/util/internal/ObjectCleaner.java

```diff
@@ -116,9 +116,9 @@ public Void run() {
             });
             cleanupThread.setName(CLEANER_THREAD_NAME);
 
-            // This Thread is not a daemon as it will die once all references to the registered Objects will go away
-            // and its important to always invoke all cleanup tasks as these may free up direct memory etc.
-            cleanupThread.setDaemon(false);
+            // Mark this as a daemon thread to ensure that we the JVM can exit if this is the only thread that is
+            // running.
+            cleanupThread.setDaemon(true);
```

Comment:
@jaymode question... does this also need to be in a privileged block ?

Reply:
This is a good question. `Thread#setDaemon` does check permissions by calling `Thread#checkAccess`, but I think it is important to look at what happens there before wrapping in a doPrivileged block. In the default [SecurityManager](https://docs.oracle.com/javase/7/docs/api/java/lang/SecurityManager.html#checkAccess(java.lang.Thread)), the access control is only enforced for system threads. In that case, I think this should not be wrapped since we are not modifying a system thread. In elasticsearch we use [SecureSM](https://github.com/elastic/securesm/blob/master/src/main/java/org/elasticsearch/SecureSM.java), which has a stricter policy but even then I don't think you'd want to elevate privileges here since modifying threads within your own threadgroup or one of its descendants is allowed.

Reply:
@jaymode thanks a lot for the answer and sorry for bothering you but after your last PR I was wondering if I am doing something stupid ;)